csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs

```diff
@@ -29,25 +31,23 @@ public CSharpAutobuildOptions(IBuildActions actions) : base(actions)
 
     public class CSharpAutobuilder : Autobuilder<CSharpAutobuildOptions>
     {
-        public CSharpAutobuilder(IBuildActions actions, CSharpAutobuildOptions options) : base(actions, options) { }
+        private const string buildCommandDocsUrl =
+            "https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-the-codeql-workflow-for-compiled-languages";
 
-        public override BuildScript GetBuildScript()
-        {
-            /// <summary>
-            /// A script that checks that the C# extractor has been executed.
-            /// </summary>
-            BuildScript CheckExtractorRun(bool warnOnFailure) =>
-                BuildScript.Create(actions =>
-                {
-                    if (actions.FileExists(Extractor.GetCSharpLogPath()))
-                        return 0;
+        private readonly AutoBuildRule autoBuildRule;
 
-                    if (warnOnFailure)
-                        Log(Severity.Error, "No C# code detected during build.");
+        private readonly DiagnosticClassifier diagnosticClassifier;
 
-                    return 1;
-                });
+        protected override DiagnosticClassifier DiagnosticClassifier => diagnosticClassifier;
```

Comment:
Explicit backing field can be avoided
```csharp
protected override DiagnosticClassifier DiagnosticClassifier { get; }
```

Reply:
Changing this led to [a ql-for-ql warning](https://github.com/github/codeql/pull/12217#discussion_r1121884214), which I think can be ignored.